File: python/FoodChain.py
from typing import List, NamedTuple

class Beast(NamedTuple):
    
    kind: str
    rhyme: str
BEASTS = [
    Beast("fly", ""),
    Beast("spider", "It wriggled and jiggled and tickled inside her."),
    Beast("bird", "How absurd to swallow a bird!"),
    Beast("cat", "Imagine that, to swallow a cat!"),
    Beast("dog", "What a hog, to swallow a dog!"),
    Beast("goat", "Just opened her throat and swallowed a goat!"),
    Beast("cow", "I don't know how she swallowed a cow!"),
    Beast("horse", "")
]

def verse(nth: int) -> List[str]:
    
    beast = BEASTS[nth-1]
    result = [f"I know an old lady who swallowed a {beast.kind}."]
    if beast.kind != "horse":
        if beast.rhyme:
            result.append(beast.rhyme)
        for other_beast in reversed(BEASTS[:nth-1]):
            roll = f"She swallowed the {beast.kind} to catch the {other_beast.kind}"
            if other_beast.kind == "spider":
                roll += " that {0}".format(other_beast.rhyme[3:-1])
            roll += "."
            result.append(roll)
            beast = other_beast
        result.append("I don't know why she swallowed the fly. Perhaps she'll die.")
    else:
        result.append("She's dead, of course!")
    return result
    
def recite(start: int, end: int) -> List[str]:
    result = []
    for nth in range(start, end + 1):
        result.extend(verse(nth))
        if nth < end:
            result.append("")
    return result

File: javascript/FoodChain.js
//
// This is only a SKELETON file for the 'Food Chain' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Song {
  verse(num) {
    return this.verses(num);
  }

  verses(from,to = from) {
       const animals = {
    1:['fly',"I don't know why she swallowed the fly. Perhaps she'll die."],
    2:['spider',"It wriggled and jiggled and tickled inside her."],
    3:['bird',"How absurd to swallow a bird!"],
    4:['cat',"Imagine that, to swallow a cat!"],
    5:['dog',"What a hog, to swallow a dog!"],
    6:['goat',"Just opened her throat and swallowed a goat!"],
    7:['cow',"I don't know how she swallowed a cow!"],
    8:['horse',"She's dead, of course!"]
  }
  let finalSong = '';
  let number = from;
  let difference = from === to;  
  do{
  finalSong = finalSong + `I know an old lady who swallowed a ${animals[number][0]}.` + '\n';
  finalSong = finalSong + animals[number][1]+'\n';
  if(number > 1 && number < 8){
  do{
    if(animals[from-1][0] === 'spider'){
      finalSong = finalSong + `She swallowed the ${animals[from][0]} to catch the ${animals[from-1][0]} that wriggled and jiggled and tickled inside her.`+'\n';
    }else{
      finalSong = finalSong + `She swallowed the ${animals[from][0]} to catch the ${animals[from-1][0]}.`+'\n';
    }
    from--;
  }
  while(from > 1);
    finalSong = finalSong + animals[1][1]+ '\n';
  } 
   if(number < to){
    finalSong = finalSong + '\n';
  }  
  number++;
  from = number;  
  }
  while(number <= to);
  difference === true ? finalSong : finalSong += '\n'  
  return finalSong;
  }
  
}